### PR TITLE
Change legacy beatmap skins to not handle `@2x` textures (to match stable)

### PR DESCRIPTION
**Pull request by frenzibyte**
- Resolves part of https://github.com/ppy/osu/issues/26239

In stable, when looking up a texture in beatmap skin, the `@2x` suffix does not mean anything and the texture is returned normally with no adjustment to scale.

This only resolves the issue above when beatmap skin is enabled. When disabled, a whole other set of issues arise (see [discord](https://discord.com/channels/188630481301012481/188630652340404224/1197101872033574932) for quick summary). PR'ing this in the mean time while I look at the rest of the issues.

### CODE REVIEW

**Comment by peppy** [comment]
So is this implying a beatmap has an `@2x` sprite which is completely different to the 1x version, and it gets used instead? Even after reading the code and OP here, I'm not sure how this fixes the related issue.

**Comment by frenzibyte** [comment]
The "new beginnings" beatmap has the gameplay-looking elements (e.g. `hitcircle`, `sliderb`) all suffixed with `@2x` and being double in size. When stable loads these elements, it loads them and display them as-is (no scale adjustment applied).

For a quick testing method, play "new beginnings" on a classic skin and make sure "beatmap skin" is turned on (when it's off it's still broken, that will be resolved later), then compare between master and PR.

master:

CleanShot 2024-01-17 at 13 39 37

PR:

CleanShot 2024-01-17 at 13 40 56

(note slider ball is still broken in both screenshots, something to address in another PR)

**Comment by peppy** [comment]
I see, thanks.

File: osu.Game.Tests/NonVisual/Skinning/LegacySkinTextureFallbackTest.cs [hunk 1]
```diff
@@ -127,8 +127,29 @@ public void TestReturnNullOnFallbackFailure()
             Assert.IsNull(texture);
         }
 
+        [Test]
```
Comment (peppy) [inline osu.Game.Tests/NonVisual/Skinning/LegacySkinTextureFallbackTest.cs:130]: Can you also add a test testing the inverse behaviour?

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game.Tests/NonVisual/Skinning/LegacySkinTextureFallbackTest.cs (hunk 1 above)
Comment (frenzibyte) [reply osu.Game.Tests/NonVisual/Skinning/LegacySkinTextureFallbackTest.cs:130]: This is supposed to be handled by most test cases above, but I'll add one for the sake of being explicit.

**Review by peppy — APPROVED** [review]

**Pull request merged**